src/core_modules/capture-core/components/WidgetsRelationship/common/RelationshipsWidget/RelationshipsWidget.component.js

```diff
@@ -31,6 +33,27 @@ const RelationshipsWidgetPlain = ({
     const { data: relationshipTypes } = useRelationshipTypes(cachedRelationshipTypes);
     const groupedLinkedEntities = useGroupedLinkedEntities(sourceId, relationshipTypes, relationships);
 
+    if (isLoading) {
+        return (
+            <Widget
+                header={(
+                    <div className={classes.header}>
+                        <span className={classes.icon}>
+                            <IconLink24 />
+                        </span>
+                    </div>
+                )}
+                onOpen={() => setOpenStatus(true)}
+                onClose={() => setOpenStatus(false)}
+                open={open}
+            >
+                <LoadingMaskElementCenter
```

Comment:
Hey,
To easily reuse the loading state for other widgets, does it make sense to move this logic within the `Widget` component? You can add a new prop (`loading` or `ready`) to the `Widget` and based on that display `LoadingMaskElementCenter` or the content of the widget. WDYT?
Thanks!